lib/auth/userloginstate/generator.go

```diff
@@ -146,6 +146,10 @@ func (g *Generator) addAccessListsToState(ctx context.Context, user types.User,
 
 		state.Spec.Roles = append(state.Spec.Roles, accessList.Spec.Grants.Roles...)
 
+		if state.Spec.Traits == nil && len(accessList.Spec.Grants.Traits) > 0 {
```

Comment:
Should we assign the new map even if the access list doesn't have traits?
This is just to prevent future cases that can cause another panic

Reply:
I figured we'd try to maintain the original `nil` just in case there are downstream consumers that rely on that for some reason.